library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/obfuscator/rules/MaskInputObfuscationRule.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.recorder.obfuscator.rules
+
+import android.widget.TextView
+import com.datadog.android.sessionreplay.internal.recorder.obfuscator.FixedLengthStringObfuscator
+
+internal class MaskInputObfuscationRule(
+    private val fixedLengthStringObfuscator: FixedLengthStringObfuscator =
+        FixedLengthStringObfuscator(),
+    private val textTypeResolver: TextTypeResolver =
+        TextTypeResolver(),
+    private val textValueResolver: TextValueResolver = TextValueResolver()
+) : TextValueObfuscationRule {
+    override fun resolveObfuscatedValue(textView: TextView): String {
+        val textType = textTypeResolver.resolveTextType(textView)
+        val textValue = textValueResolver.resolveTextValue(textView)
+        return when (textType) {
+            TextType.SENSITIVE_TEXT -> fixedLengthStringObfuscator.obfuscate(textValue)
+            TextType.INPUT_TEXT -> fixedLengthStringObfuscator.obfuscate(textValue)
+            TextType.OPTION_TEXT -> fixedLengthStringObfuscator.obfuscate(textValue)
```

Comment:
question/ Can `OPTION_TEXT` appear in `TextView` element? From `TextTypeResolver` implementation I see it can not. It can mean we lack cohesion in this design.

Reply:
Not yet, coming in a new PR where like explained above I am going to use the `TextView#parent` property to check for its container.